### PR TITLE
[IOTDB-3656] mpp load

### PR DESCRIPTION
**Pull request by yschengzi**
design doc:https://apache-iotdb.feishu.cn/docx/doxcnyBYWzek8ksSEU6obZMpYLe

un-supported: last-cache, modification, clean up, tsfile name restriction

### CODE REVIEW

**Comment by wangchao316** [comment]
?

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/scheduler/load/LoadTsFileScheduler.java [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+public class LoadTsFileScheduler implements IScheduler {
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/scheduler/load/LoadTsFileScheduler.java:57]: Add javadoc?

File: server/src/main/java/org/apache/iotdb/db/engine/load/LoadTsFileManager.java [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class LoadTsFileManager {
```
[the remark this answers is not in the document]
Comment (yschengzi) [reply server/src/main/java/org/apache/iotdb/db/engine/load/LoadTsFileManager.java:48]: done.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/StandaloneSchemaFetcher.java [hunk 3]
```diff
@@ -125,7 +125,7 @@ public ISchemaTree fetchSchemaListWithAutoCreate(
       int totalSize = 0;
       boolean isAligned = isAlignedList.get(entry.getValue().get(0));
       for (int index : entry.getValue()) {
-        if (isAlignedList.get(entry.getValue().get(index)) != isAligned) {
+        if (isAlignedList.get(index) != isAligned) {
```
[the remark this answers is not in the document]
Comment (yschengzi) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/StandaloneSchemaFetcher.java:128]: this is a bug in schema, i discuss it with zyk.

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/DataRegion.java [hunk 4]
```diff
@@ -2659,11 +2659,11 @@ private void tryToDeleteLastCacheByDevice(PartialPath deviceId) {
...
-    try {
-      IoTDB.schemaProcessor.deleteLastCacheByDevice(deviceId);
-    } catch (MetadataException e) {
-      // the path doesn't cache in cluster mode now, ignore
-    }
+    //    try {
+    //      IoTDB.schemaProcessor.deleteLastCacheByDevice(deviceId);
+    //    } catch (MetadataException e) {
+    //      // the path doesn't cache in cluster mode now, ignore
+    //    }
```
[the remark this answers is not in the document]
Comment (yschengzi) [reply server/src/main/java/org/apache/iotdb/db/engine/storagegroup/DataRegion.java:2666]: last cache will be supported in future pr. may be a TODO here is better.

File: server/src/main/java/org/apache/iotdb/db/engine/load/LoadTsFileManager.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (yschengzi) [reply server/src/main/java/org/apache/iotdb/db/engine/load/LoadTsFileManager.java:48]: done.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/planner/plan/node/load/LoadTsFileNode.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  }
+
+  @Override
+  public TRegionReplicaSet getRegionReplicaSet() {
+    return null;
+  }
+
+  @Override
+  public List<PlanNode> getChildren() {
+    return null;
```
[the remark this answers is not in the document]
Comment (yschengzi) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/planner/plan/node/load/LoadTsFileNode.java:60]: `InsertTabletNode` also return null

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/planner/plan/node/load/LoadTsFileNode.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+  }
+
+  @Override
+  public int allowedChildCount() {
+    return NO_CHILD_ALLOWED;
+  }
+
+  @Override
+  public List<String> getOutputColumnNames() {
+    return null;
```
[the remark this answers is not in the document]
Comment (yschengzi) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/planner/plan/node/load/LoadTsFileNode.java:78]: same above.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/scheduler/load/LoadTsFileScheduler.java (hunk 1 above)
Comment (yschengzi) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/scheduler/load/LoadTsFileScheduler.java:57]: done.

**Review by HTHou — APPROVED** [review]

File: integration-test/src/main/java/org/apache/iotdb/it/env/AbstractNodeWrapper.java [hunk 7]
```diff
@@ -174,7 +174,9 @@ public void start() {
               "-Xms200m",
               "-Xmx200m",
               "-XX:MaxDirectMemorySize=200m",
-              "-Djdk.nio.maxCachedBufferSize=262144"));
+              "-Djdk.nio.maxCachedBufferSize=262144",
+              "-cp",
+              templateNodeLibPath));
```
[the remark this answers is not in the document]
Comment (yschengzi) [reply integration-test/src/main/java/org/apache/iotdb/it/env/AbstractNodeWrapper.java:179]: IT does not work well on my computer unless add this.

[the remark this answers is not in the document]
Comment (yschengzi) [reply integration-test/src/main/java/org/apache/iotdb/it/env/AbstractNodeWrapper.java:179]: for more discuss, can see link: https://apacheiotdb.slack.com/archives/C02JWGWE49Y

File: integration-test/src/main/java/org/apache/iotdb/itbase/env/BaseConfig.java [hunk 8]
```diff
@@ -70,6 +70,10 @@ default BaseConfig setDataRegionNum(int dataRegionNum) {
     return this;
   }
 
+  default boolean isEnablePartition() {
+    return true;
```
[the remark this answers is not in the document]
Comment (yschengzi) [reply integration-test/src/main/java/org/apache/iotdb/itbase/env/BaseConfig.java:74]: i will change it in next pr

**Pull request merged**